nncf/quantization/algorithms/post_training/algorithm.py

```diff
@@ -160,31 +160,17 @@ def available_backends(self) -> Dict[str, BackendType]:
         return
 
     def get_statistic_points(self, model: TModel, graph: NNCFGraph) -> StatisticPointsContainer:
-        return StatisticPointsContainer()
-
-    def _create_statistics_aggregator(self, dataset: Dataset, backend: BackendType) -> StatisticsAggregator:
-        """
-        Creates backend-specific StatisticsAggregator.
-
-        :param engine: Engine for the model execution
-        :param dataset: Dataset for the statistics collection and validation
-        :param model_transformer: Backend-specific ModelTransformerBase instance
-        :param backend: Model backend type for the further differentiations
-        :return: Backend-specific StatisticsAggregator
-        """
-        if backend == BackendType.ONNX:
-            from nncf.onnx.statistics.aggregator import ONNXStatisticsAggregator
-
-            return ONNXStatisticsAggregator(dataset)
-        if backend == BackendType.OPENVINO:
-            from nncf.openvino.statistics.aggregator import OVStatisticsAggregator
-
-            return OVStatisticsAggregator(dataset)
-        if backend == BackendType.TORCH:
-            from nncf.torch.statistics.aggregator import PTStatisticsAggregator
+        if self.first_stage_algorithms:
+            raise NotImplementedError(
+                "Statistic points are not supported yet for SmoothQuant and ChannelAlignment algorithms."
+            )
 
-            return PTStatisticsAggregator(dataset)
-        return None
+        output = StatisticPointsContainer()
+        for algorithm in self.algorithms:
+            for statistic_points in algorithm.get_statistic_points(model, graph).values():
+                for statistic_point in statistic_points:
+                    output.add_statistic_point(statistic_point)
+        return output
```

Comment:
I restored behavior that was before https://github.com/openvinotoolkit/nncf/pull/2011.
The task (ref: 117471) was created to resolve issues.